azure.yaml

```diff
@@ -8,16 +8,30 @@ services:
     project: ./app/backend
     language: py
     host: appservice
+  scripts:
+    project: ./scripts
+    language: py
+    host: ""
 hooks:
     postprovision:
-      shell: pwsh
-      run: ./scripts/prepdocs.ps1
-      interactive: true
-      continueOnError: false
+      windows:
+        shell: pwsh
+        run: ./scripts/prepdocs.ps1
+        interactive: true
+        continueOnError: false
+      posix:
+        shell: sh
+        run: scripts/scripts_env/bin/python scripts/prepdocs.py 'data/*' --storageaccount $AZURE_STORAGE_ACCOUNT --container $AZURE_STORAGE_CONTAINER --searchservice $AZURE_SEARCH_SERVICE --index $AZURE_SEARCH_INDEX -v
+        interactive: true
+        continueOnError: false
     predeploy:
+      windows:
         shell: pwsh
         run:  cd ./app/frontend;npm install;npm run build
         interactive: true
         continueOnError: false
-
-      
+      posix:
+        shell: sh
+        run:  cd ./app/frontend;npm install;npm run build
+        interactive: true
+        continueOnError: false
```

Comment:
Why dup?

Reply:
Different shells. Is there a better way to do it?

Reply:
The default section will default to appropriate shell for the detected OS.

Reply:
That does not seem to be the case:

```sh
$> azd deploy

ERROR: failed running pre hooks: failed running scripts for hooks 'azd deploy', hook configuration for 'predeploy' is invalid, unable to determine script type. Ensure 'Shell' parameter is set in configuration options
```

Reply:
I was mistaken.